articles/clients/client-grant-types.md

```diff
@@ -0,0 +1,135 @@
+---
+  description: Using the Grant Types property on Clients
+  toc: true
+---
+
+# Client Grant Types
+
+Auth0 provides many authentication and authorization flows that suit your needs, and depending on your use case, you may wish to limit the use of certain grant types for a particular Client. Auth0 includes a `grant_type` property on each Client for this purpose.
+
+## Grant Types
+
+The following is a list of grant types valid for Auth0 Clients. Each of the items represents one of the three types of authorization flows.
+
+There are three possible types of authorization flows:
+
+* Auth0 extension;
+* Auth0 legacy;
+* Specification-conformant (such as OpenID Connect);
+
+The following is a list of Auth0 extension/OIDC specification-conformant `grant_types` that currently exist for use in Auth0:
+
+* `implicit`
+* `authorization_code`
+* `client_credentials`
+* `password`
+* `refresh_token`
+* `http://auth0.com/oauth/grant-type/password-realm`
+* `http://auth0.com/oauth/grant-type/mfa-oob`
+* `http://auth0.com/oauth/grant-type/mfa-otp`
+* `http://auth0.com/oauth/grant-type/mfa-recovery-code`
+
+The following is a list of legacy `grant_types`:
+
+* `http://auth0.com/oauth/legacy/grant-type/ro`
+* `http://auth0.com/oauth/legacy/grant-type/ro/jwt-bearer`
+* `http://auth0.com/oauth/legacy/grant-type/delegation/refresh_token`
+* `http://auth0.com/oauth/legacy/grant-type/delegation/id_token`
+* `http://auth0.com/oauth/legacy/grant-type/access_token`
+
+## Edit the `grant_type` Property
+
+To edit the `grant_type` property for your Auth0 Client, you'll need to make the appropriate call to the [Update a Client endpoint](/api/management/v2#!/Clients/patch_clients_by_id) of the [Management API](/api/management/v2).
+
+:::panel-warning Legacy Grant Types
+As of TBD, new Auth0 customers **cannot** add *any* of the legacy grant types to their Clients. Only customers as of TBD can add legacy grant types to their existing Clients.
+:::
+
+Auth0 requires the `grant_type` property. Attempting to use *any* flow with a Client lacking the appropriate `grant_type` for that flow (or with the field empty) will result in the following error:
+
+```text
+Grant type `grant_type` not allowed for the client.
+```
+
+### Existing Clients
+
+To avoid changes in functionality for current Auth0 customers, we will populate the `grant_types` property for all existing Clients as of TBD with **all** Auth0 legacy, Auth0 extension, and specification-conformant grant types.
+
+### New Clients
+
+Depending on whether a newly-created Client is **public** or **confidential**, the Client will have varying access to grant types.
+
+### Public Clients
+
+Public Clients, indicated by the `token_endpoint_auth_method` flag set to `none`, are those created in the Dashboard for Native and Single Page Applications. They can be set to the following grant types:
```

Comment:
Public Clients, indicated by the `token_endpoint_auth_method` flag set to `none`, are those created in the Dashboard for Native and Single Page Applications. By default, Public Clients are created with the following `grant_types`: